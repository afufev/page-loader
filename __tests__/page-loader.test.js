import os from 'os';
import path from 'path';
import url from 'url';
import { promises as fs } from 'fs';
import nock from 'nock';
import axios from 'axios';
import httpAdapter from 'axios/lib/adapters/http';

import pageLoader from '../src';

nock.disableNetConnect();

axios.defaults.adapter = httpAdapter;

let output;
let expectedHtml;
let dest;

const fixturesPathExpectedHtml = './__tests__/__fixtures__/test-hexlet-io-courses.html';
const host = 'http://ru.hexlet.io/';
const filename = 'ru-hexlet-io-courses.html';
const testPath = '/courses';
const address = url.resolve(host, testPath);


describe('step1', () => {
  beforeEach(async () => {
    output = await fs.mkdtemp(path.resolve(os.tmpdir(), 'page-loader-'));
    expectedHtml = await fs.readFile(fixturesPathExpectedHtml, 'utf-8');
    dest = path.join(output, filename);
  });
  it('#get', async () => {
    nock(host).get('/courses').reply(200, expectedHtml);
    await pageLoader(address, output);
    const response = await fs.readFile(dest, 'utf-8');
    expect(response).toBe(expectedHtml);
  });
  it('#error', async () => {
    try {
      await pageLoader('unknown', output);
    } catch (e) {
      expect(e).toBeInstanceOf(Error);
    }
  });
});
