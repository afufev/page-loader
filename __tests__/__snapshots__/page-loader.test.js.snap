// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error handling #404: not found 1`] = `"Unable to download the page. Error: EADDRNOTAVAIL: recieved status 404 - null: address not available"`;

exports[`error handling #443: no host 1`] = `"Unable to download the page. Error: ENOTFOUND: recieved status 443 connecting to unknown"`;

exports[`error handling #ENOENT 1`] = `"Unable to download the page. Error: ENOENT: no such file or directory, mkdir 'unknown/ru-hexlet-io-courses_files'"`;

exports[`error handling #unknown error 1`] = `"Unable to download the page. Unknown error, please, send us report"`;

exports[`load page and resources #getBody 1`] = `
"<!DOCTYPE html><html><head>
    <meta charset=\\"utf-8\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <title>Hexlet</title>
    <link href=\\"ru-hexlet-io-courses_files/assets-application.css\\" type=\\"text/css\\" rel=\\"stylesheet\\">
    <link href=\\"http://ru.hexlet.io/courses\\" rel=\\"canonical\\">
    <script src=\\"ru-hexlet-io-courses_files/assets-script.js\\"></script>
  </head>
  <body>
    <nav>
      <img src=\\"ru-hexlet-io-courses_files/assets-hexlet-logo-e99fc2b3b7c1eec88899f3af1435a39aaac6fd29d011dfe2342499c0884b7a96.png\\" alt=\\"hexlet-logo\\">
      <ul>
        <li><a href=\\"https://ru.hexlet.io/my\\">My Hexlet</a></li>
        <li><a href=\\"https://ru.hexlet.io/professions\\">professions</a></li>
        <li><a href=\\"https://ru.hexlet.io/courses\\">All courses</a></li>
        <li><a href=\\"https://ru.hexlet.io/blog\\">Blog</a></li>
        <li><a href=\\"https://ru.hexlet.io/knowledge\\">Database</a></li>
        <li><a href=\\"https://ru.hexlet.io/community\\">Community</a></li>
      </ul>
    </nav>
    <section>
      <ul>
        <li><a href=\\"#\\">Web-development</a></li>
        <li><a href=\\"#\\">Learning Russian</a></li>
        <li><a href=\\"#\\">Linux: users</a></li>
        <li><a href=\\"#\\">JS: Trees</a></li>
        <li><a href=\\"#\\">JS: Functions</a></li>
        <li><a href=\\"#\\">bash</a></li>
        <li><a href=\\"#\\">PHP: Functions</a></li>
        <li><a href=\\"#\\">SQL</a></li>
      </ul>
    </section>
    <footer>
      <p> Hexlet 2019</p>
    </footer>
  

</body></html>"
`;
